code/modules/client/client_procs.dm

```diff
@@ -371,70 +372,72 @@ GLOBAL_LIST(external_rsc_urls)
 /client/Destroy()
 	return QDEL_HINT_HARDDEL_NOW
 
-/client/proc/set_client_age_from_db()
+/client/proc/set_client_age_from_db(connectiontopic)
 	if (IsGuestKey(src.key))
 		return
-
 	if(!SSdbcore.Connect())
 		return
-
 	var/sql_ckey = sanitizeSQL(src.ckey)
-
-	var/datum/DBQuery/query_get_client_age = SSdbcore.NewQuery("SELECT id, datediff(Now(),firstseen) as age FROM [format_table_name("player")] WHERE ckey = '[sql_ckey]'")
-	if(!query_get_client_age.Execute())
-		return
-
-	while(query_get_client_age.NextRow())
-		player_age = text2num(query_get_client_age.item[2])
-		return
-
-	//no match mark it as a first connection for use in client/New()
-	player_age = -1
-
-
-/client/proc/sync_client_with_db(connectiontopic)
-	if (IsGuestKey(src.key))
-		return
-
-	if (!SSdbcore.Connect())
-		return
-
-	var/sql_ckey = sanitizeSQL(ckey)
-
-	var/datum/DBQuery/query_get_ip = SSdbcore.NewQuery("SELECT ckey FROM [format_table_name("player")] WHERE ip = INET_ATON('[address]') AND ckey != '[sql_ckey]'")
-	query_get_ip.Execute()
-	related_accounts_ip = ""
-	while(query_get_ip.NextRow())
-		related_accounts_ip += "[query_get_ip.item[1]], "
-
-	var/datum/DBQuery/query_get_cid = SSdbcore.NewQuery("SELECT ckey FROM [format_table_name("player")] WHERE computerid = '[computer_id]' AND ckey != '[sql_ckey]'")
-	if(!query_get_cid.Execute())
-		return
-	related_accounts_cid = ""
-	while (query_get_cid.NextRow())
-		related_accounts_cid += "[query_get_cid.item[1]], "
-
 	var/admin_rank = "Player"
 	if (src.holder && src.holder.rank)
 		admin_rank = src.holder.rank.name
 	else
 		if (check_randomizer(connectiontopic))
 			return
-
-	var/sql_ip = sanitizeSQL(src.address)
-	var/sql_computerid = sanitizeSQL(src.computer_id)
+	var/sql_ip = sanitizeSQL(address)
+	var/sql_computerid = sanitizeSQL(computer_id)
 	var/sql_admin_rank = sanitizeSQL(admin_rank)
-
-
-	var/datum/DBQuery/query_log_player = SSdbcore.NewQuery("INSERT INTO [format_table_name("player")] (id, ckey, firstseen, lastseen, ip, computerid, lastadminrank) VALUES (null, '[sql_ckey]', Now(), Now(), INET_ATON('[sql_ip]'), '[sql_computerid]', '[sql_admin_rank]') ON DUPLICATE KEY UPDATE lastseen = VALUES(lastseen), ip = VALUES(ip), computerid = VALUES(computerid), lastadminrank = VALUES(lastadminrank)")
-	if(!query_log_player.Execute())
+	var/new_player
+	var/datum/DBQuery/query_client_in_db = SSdbcore.NewQuery("SELECT 1 FROM [format_table_name("player")] WHERE ckey = '[sql_ckey]'")
+	if(!query_client_in_db.Execute())
 		return
-
-	//Logging player access
-
+	if(!query_client_in_db.NextRow())
+		new_player = 1
+		account_join_date = findJoinDate()
+		var/datum/DBQuery/query_add_player = SSdbcore.NewQuery("INSERT INTO [format_table_name("player")] VALUES ('[sql_ckey]', Now(), Now(), INET_ATON('[sql_ip]'), '[sql_computerid]', '[sql_admin_rank]', account_join_date ? \'[account_join_date]\' : NULL)")
```

Comment:
is it theoretically possible for someone, especially an admin, to edit the account_join date before it gets inserted? if so they could sql inject attack.